services/enqueueing/enqueueing.go

```diff
@@ -0,0 +1,45 @@
+package enqueueing
+
+import (
+	"errors"
+
+	"google-scraper/database"
+	"google-scraper/models"
+
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/gocraft/work"
+	"github.com/beego/beego/v2/server/web"
+)
+
+var enqueuer *work.Enqueuer
+
+func init() {
+	enqueuer = work.NewEnqueuer("google_scraper", database.GetRedisPool())
+}
+
+func DelayedEnqueue(keyword *models.Keyword, keywordIndex int64) (*work.ScheduledJob, error) {
```

Comment:
In terms of naming this can be improved. `Delayed` kid of mix two concepts: asynchronicity and throttling. Also, `keywordIndex` is too coupled with the fact that the keywords are enqueued in a loop. Instead I recommend the following:

```suggestion
func EnqueueKeywordJob(keyword *models.Keyword, throttleMultiplier int64) (*work.ScheduledJob, error) {
```

Reply:
Fixed it https://github.com/junan/google-scraper/pull/33/commits/8fbdb9fdeeed16bb3f4a1d2badad0c7a4eb89fa1